cmd/cdc/server/server.go

```diff
@@ -206,6 +214,83 @@ func (o *options) getCredential() *security.Credential {
 	}
 }
 
+func parseConfigFlagFromOSArgs() string {
+	var serverConfigFilePath string
+	for i, arg := range os.Args[1:] {
+		if strings.HasPrefix(arg, "--config=") {
+			serverConfigFilePath = strings.SplitN(arg, "=", 2)[1]
+		} else if arg == "--config" && i+2 < len(os.Args) {
+			serverConfigFilePath = os.Args[i+2]
+		}
+	}
+	log.Info("parse config file path from os.Args", zap.String("config", serverConfigFilePath))
+
+	// If the command is `cdc cli changefeed`, means it's not a server config file.
+	if slices.Contains(os.Args, "cli") && slices.Contains(os.Args, "changefeed") {
+		serverConfigFilePath = ""
+	}
+
+	return serverConfigFilePath
+}
+
+func isNewArchEnabledByConfig(serverConfigFilePath string) bool {
+	cfg := config.GetDefaultServerConfig()
+	if len(serverConfigFilePath) > 0 {
+		// strict decode config file, but ignore debug item
+		if err := util.StrictDecodeFile(serverConfigFilePath, "TiCDC server", cfg, config.DebugConfigurationItem); err != nil {
+			log.Error("failed to parse server configuration, please check the config file for errors and try again.", zap.Error(err))
+			return false
+		}
+	}
+
+	return cfg.Newarch
+}
+
+func isNewArchEnabled(o *options) bool {
+	newarch := o.serverConfig.Newarch
+	if newarch {
+		log.Info("Set newarch from command line")
```

Comment:
These logs should not be needed, because it rarely cause any confusion that how new arch is enabled.

Reply:
OK, I'll change these to debug log.